### PR TITLE
Upgrade to supoprt new Auro design tokens #94

### PR DESCRIPTION
**Pull request by blackfalcon**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #94 

## Summary:

This commit updates all the token references to support the new design token naming spec. All CSS also supports fallback values to ensure a smooth transition from legacy to new.

TODO: There are 2 to-dos in this commit. Once there is an assumption that most teams have transitioned to the new token spec, the code that these to-dos are referencing can be removed as well as any npm package or reference to the legacy tokens.

Changes to be committed:
modified:   package-lock.json
modified:   package.json
modified:   src/alaskaStyle.scss
modified:   src/iconStyle.scss
modified:   src/style.scss

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,50 +21,51 @@
...
   },
   "devDependencies": {
-    "@alaskaairux/design-tokens": "^3.13.0",
-    "@alaskaairux/icons": "^4.25.0",
-    "@aurodesignsystem/webcorestylesheets": "^4.3.0",
-    "@commitlint/cli": "^17.6.1",
-    "@commitlint/config-conventional": "^17.6.1",
-    "@open-wc/testing": "^3.1.8",
-    "@rollup/plugin-node-resolve": "^15.0.2",
+    "@alaskaairux/design-tokens": "^3.15.5",
```
Comment (jason-capsule42) [inline package.json:32]: Do we need to keep the old tokens around?

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (blackfalcon) [reply package.json:32]: See lines https://github.com/AlaskaAirlines/auro-icon/pull/95/files#diff-c0a5ddd5b365322730802e8cb15c95c3b6e965cbd3e569c69a2f089753844cdbR13-R18

Once we release the updated auro icons and update this repo with the new version, then we can delete.

File: src/iconStyle.scss [hunk 2]
```diff
@@ -3,82 +3,91 @@
...
+
 @import "./../node_modules/@aurodesignsystem/webcorestylesheets/dist/auroElement/auroElement";
 
 :host {
   display: inline-block;
   line-height: unset;
 
   .primary {
-    color: var(--auro-color-icon-primary-on-light);
+    color: var(--auro-color-icon-primary-default, $ds-color-icon-primary-default);
```
[the remark this answers is not in the document]
Comment (blackfalcon) [reply src/iconStyle.scss:16]: lol... this why I shouldn't be coding at midnight.

**Review by jason-capsule42 — DISMISSED** [review]

File: src/alaskaStyle.scss [hunk 3]
```diff
@@ -3,12 +3,21 @@
...
 
 :host {
   display: inline-block;
+  // listing both legacy and newer tokens to support
+  // transition to new naming spec.
+  //
+  // TO DO: Remove --auro token when confident
+  // that new spec has been adopted
   --auro-size-lg: 100%;
+  --ds-size-300: 100%;
```
Comment (jordanjones243) [inline src/alaskaStyle.scss:20]: What are these two variables for? I do not see them being used in the code.

File: src/iconStyle.scss [hunk 4]
```diff
@@ -3,82 +3,91 @@
...
 }
 
 :host([customSize]) {
+  // listing both legacy and newer tokens to support
+  // transition to new naming spec.
+  //
+  // TO DO: Remove --auro token when confident
+  // that new spec has been adopted
   --auro-size-lg: 100%;
+  --ds-size-300: 100%;
```
Comment (jordanjones243) [inline src/iconStyle.scss:90]: Same comment as above.

File: src/alaskaStyle.scss (hunk 3 above)
Comment (blackfalcon) [reply src/alaskaStyle.scss:20]: See comment note

```
  // TO DO: Remove --auro token when confident
  // that new spec has been adopted
```

These are both there to support a fallback scenario until everyone has moved to the new `-ds` tokens.

File: src/iconStyle.scss (hunk 4 above)
Comment (blackfalcon) [reply src/iconStyle.scss:90]: Again... 

```
  // TO DO: Remove --auro token when confident
  // that new spec has been adopted
```

**Review by jordanjones243 — DISMISSED** [review]

File: package.json [hunk 5]
```diff
@@ -81,7 +80,8 @@
     ]
   },
   "scripts": {
-    "build": "npm-run-all build:sass sass:render cssLint dist:js bundler postinstall build:api",
+    "build": "npm-run-all build:sass sass:render cssLint bundler postinstall build:api",
```
Comment (jordanjones243) [inline package.json:83]: `types` needs to be added into the build script

**Review by jordanjones243 — CHANGES_REQUESTED** [review]

**Review by jordanjones243 — APPROVED** [review]

**Pull request merged**